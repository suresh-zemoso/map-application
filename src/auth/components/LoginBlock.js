import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { TextField } from '@material-ui/core';
import useInput from '../../hooks/useInput';
import { useDispatch, useSelector } from "../../hooks/reduxHooks";
import { login } from '../../auth/actions/loginAction';
import { withRouter } from 'react-router';
import Alert from '@material-ui/lab/Alert';
import PropTypes from 'prop-types';


const useStyles = makeStyles({
    root: {
        position: 'absolute',
        top: '15%',
        display: ' flex',
        justifyContent: 'center',
        width: '60%',
        height: '40%',
    },
    buttonRoot: {
        width: '100%',
        margin: '3%',
        height: '2.5rem'
    },
    inputField: {
        width: '100%',
        margin: '3%',
    },
    alert: {
        margin: '3%',
        width: '100%',
        boxSizing: 'border-box'
    }

});

const defaultProps = {

}

const propTypes = {


}

const LoginBlock = (props) => {

    const classes = useStyles(props);
    const username = useInput("");
    const password = useInput("");

    const dispatch = useDispatch();
    const authentication = useSelector(state => state.authentication)

    function submitForm() {
        const credentials = {
            username: username.value,
            password: password.value,
        }
        dispatch(login(credentials))
            .then(() => { props.history.push("/locations/new") },
                (error) => {
                    console.warn(error);
                });
    }

    return (
        <div className={classes.root}>
            <div className={classes.form}>
                {authentication.error && <Alert className={classes.alert} severity="error">{authentication.errorMessage}</Alert>}
                <TextField
                    id="username-00"
                    label="Username"
                    variant="outlined"
                    required
                    classes={{ root: classes.inputField }}
                    {...username}
                />
                <TextField
                    id="password--00"
                    label="Password"
                    type="password"
                    variant="outlined"
                    required
                    classes={{ root: classes.inputField }}
                    {...password}
                />
                <Button
                    id="login-button"
                    classes={{ root: classes.buttonRoot }}
                    disabled={(username.value == "" || password.value == "") ? true : false}
                    variant="contained"
                    color="primary"
                    type="submit"
                    onClick={submitForm}>
                    Sign In
                </Button>
            </div>
        </div >
    )
}

LoginBlock.propTypes = propTypes;
LoginBlock.defaultProps = defaultProps;

export default withRouter(LoginBlock);
