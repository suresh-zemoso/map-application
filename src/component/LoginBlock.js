import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { TextField } from '@material-ui/core';
import useInput from '../hooks/useInput';
import { useDispatch } from "react-redux";
import { login } from '../actions/actions';
import { withRouter } from 'react-router';
import { loginService } from '../services/loginService';
import Alert from '@material-ui/lab/Alert';


const useStyles = makeStyles({
    root: {
        position: 'absolute',
        top: '15%',
        display: ' flex',
        justifyContent: 'center',
        width: '60%',
        height: '40%',
    },
    buttonRoot: {
        width: '100%',
        margin: '3%',
        height: '56px'
    },
    inputField: {
        width: '100%',
        margin: '3%',
    },
    alert: {
        margin: '3%',
        width: '100%',
        boxSizing: 'border-box'
    }

});

const LoginBlock = (props) => {

    const classes = useStyles(props);
    const username = useInput("");
    const password = useInput("");
    const [isBadEntry, setBadEntry] = useState(false);

    const dispatch = useDispatch();

    function submitForm() {
        const data = {
            username: username.value,
            password: password.value,
        }
        loginService.signIn(data)
            .then(response => {
                response.response && response.response.status == "401" && setBadEntry(true);
                dispatch(login(response.data.access_token))
                props.history.push("/locations/new");
            })
            .catch(error => {
                console.log(error)
            })
    }

    return (
        <div className={classes.root}>
            <div className={classes.form}>
                {isBadEntry && <Alert className={classes.alert} severity="error" onClose={() => { setBadEntry(false) }}>Incorrect username or password.</Alert>}
                <TextField
                    id="username-00"
                    label="Username"
                    variant="outlined"
                    required
                    classes={{ root: classes.inputField }}
                    {...username}
                />
                <TextField
                    id="password--00"
                    label="Password"
                    type="password"
                    variant="outlined"
                    required
                    classes={{ root: classes.inputField }}
                    {...password}
                />
                <Button
                    classes={{ root: classes.buttonRoot }}
                    disabled={(username.value == "" || password.value == "") ? true : false}
                    variant="contained"
                    color="primary"
                    type="submit"
                    onClick={() => { submitForm() }}>
                    Sign In
                </Button>
            </div>
        </div>
    )
}


export default withRouter(LoginBlock);
