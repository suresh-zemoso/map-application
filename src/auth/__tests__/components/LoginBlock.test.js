import React from 'react';
import { mount } from 'enzyme';
import LoginBlock from '../../components/LoginBlock';
import { MemoryRouter } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as ReactReduxHooks from "../../hooks/reduxHooks"


describe("Login Block", () => {
    let wrapper;
    let store;

    beforeEach(() => {
        /* mocking store */
        store = configureStore([thunk])({
            authentication: { error: true, errorMessage: 'adfkadskfajskd ' }
        });

        jest
            .spyOn(ReactReduxHooks, "useSelector")
            .mockImplementation(state => store.getState());
        /* mocking useDispatch on our mock store  */
        jest
            .spyOn(ReactReduxHooks, "useDispatch")
            .mockImplementation(() => store.dispatch);

        /* shallow rendering */
        wrapper = mount(<MemoryRouter><LoginBlock store={store} /></MemoryRouter>);
    });

    it('should render the LoginBlock Component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Click on login button should call dispatch", () => {
        let loginButton = wrapper.find(Button);
        loginButton.at(0).simulate('click');
        expect(ReactReduxHooks.useDispatch.mock.calls.length).toEqual(1);
    });
});
