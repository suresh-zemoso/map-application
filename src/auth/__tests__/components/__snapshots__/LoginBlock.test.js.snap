// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Block should render the LoginBlock Component correctly 1`] = `ReactWrapper {}`;
